Jenkinsfile.trigger

```
@@ -102,69 +102,98 @@ if (breakEarly) { return } // thanks Jenkins...
 
 // new data to be added to the past-jobs.json
 // { lastTime: unixTimestamp, url: "" }
-def buildCompletionData = [:]
+buildCompletionData = [:]
 
-for (buildObj in queue) {
-	stage(buildObj.identifier) {
-		//def json = writeJSON(json: buildObj, returnText: true)
-		//echo(json) // for debugging/data purposes
+// list of closures that we can use to wait for the jobs on.
+def waitQueue = [:]
+def addToWait(identifier, buildId, externalizableId) {
+	return {
+		// "catchError" to set "stageResult" :(
+		catchError(message: 'Build of "' + identifier + '" failed', buildResult: 'UNSTABLE', stageResult: 'FAILURE') {
+			stage(identifier) {
```

I know you tested and it _should_ work in this order, but I think it's slightly clearer if we swap these two (both so we don't catch `stage()` itself somehow failing and so it's clear our stage happens and then we set the state of it from within not without):

```suggestion
		stage(identifier) {
			// "catchError" to set "stageResult" :(
			catchError(message: 'Build of "' + identifier + '" failed', buildResult: 'UNSTABLE', stageResult: 'FAILURE') {
```

---

Seems sane. I should also move it inside the `stage(buildObj.identifier)` on https://github.com/docker-library/meta-scripts/pull/99/files#diff-a1eafb3a911bf01d23222ee7808d97d5f227a9608c46843ae4908cdca3408234R139, but that means I'll need to duplicate it for the `else` block but that seems fine.